tests/unit/backend/layers/business/test_business.py

```diff
@@ -856,6 +859,61 @@ def test_add_dataset_to_unpublished_collection_ok(self):
 
         self.step_function_provider.start_step_function.assert_called_once()
 
+    def test_reingest_published_dataset(self):
+        """A cellxgene public dataset url can be used to ingest a new dataset version."""
+
+        collection, revision = self.initialize_collection_with_an_unpublished_revision(num_datasets=1)
+        dataset_version = revision.datasets[0]
+        url = f"https://dataset_assets_domain/{dataset_version.version_id}.h5ad"
+
+        new_dataset_version_id, _ = self.business_logic.ingest_dataset(
+            revision.version_id, url, None, dataset_version.version_id
+        )
+        new_dataset_version = self.database_provider.get_dataset_version(new_dataset_version_id)
+        self.assertIsNotNone(new_dataset_version)
+        self.assertIsNone(new_dataset_version.metadata)
+        self.assertEqual(new_dataset_version.collection_id, revision.collection_id)
+        self.assertEqual(new_dataset_version.status.upload_status, DatasetUploadStatus.WAITING)
+        self.assertEqual(new_dataset_version.status.processing_status, DatasetProcessingStatus.INITIALIZED)
+        self.step_function_provider.start_step_function.assert_called_once_with(
+            revision.version_id,
+            new_dataset_version_id,
+            f'{{"anndata":"s3://artifacts/{dataset_version.version_id}/raw.h5ad","atac_fragment":null}}',
+        )
+
+    def test_reingest_published_dataset__not_h5ad(self):
+        """A cellxgene public dataset url used for reingesting an h5ad must be an h5ad"""
+        collection, revision = self.initialize_collection_with_an_unpublished_revision(num_datasets=1)
+        dataset_version = revision.datasets[0]
+        url = f"https://dataset_assets_domain/{dataset_version.version_id}.rds"
+        with self.assertRaises(InvalidIngestionManifestException):
+            self.business_logic.ingest_dataset(revision.version_id, url, None, dataset_version.version_id)
```

Comment:
This is optional, but I think two meaningful improvements that could be made to these tests are:

* Checking text of the error to make sure the correct helpful message is given
* Ideally I would check that this message is actually returned in the response, e.g. make sure the curator can actually see the helpful error message